libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAbstractPasscodeViewController.m

```diff
@@ -150,11 +152,20 @@ - (void)viewDidAppear:(BOOL)animated {
 - (void)setupPasscode:(NSString *)passcode
 {
     dispatch_async(dispatch_get_main_queue(), ^{
-        self.remainingAttempts = kMaxNumberofAttempts;
+        self.remainingAttempts = (_viewConfig.maxNumberOfAttempts) ? _viewConfig.maxNumberOfAttempts : kMaxNumberofAttempts;
         [[SFPasscodeManager sharedManager] changePasscode:passcode];
         [SFSecurityLockout setupTimer];
         SFSecurityLockoutAction action = [self controllerModeToLockoutAction];
-        [SFSecurityLockout unlock:YES action:action passcodeConfig:self.configData];
+        
+        if ([SFSecurityLockout passcodeLength] == 0) {
```

Comment:
When could it be zero?

Reply:
Passcode length will be 0 when the actual passcode length is unknown. This only happens as a one time upgrade step when transitioning from the old UI where we effectively stored minimum length (because it did not matter then) to the new UI where actual passcode length is necessary.

I will create a bool and add comments to make this clear.